dockstore-launcher/src/main/java/io/github/collaboratory/LauncherCWL.java

```diff
@@ -251,27 +252,37 @@ private void prepUploadsHelper(Map<String, Object> inputsOutputs, final Map<Stri
             String cwlID) {
         // now that I have an input name from the CWL I can find it in the JSON parameterization for this run
         LOG.info("JSON: {}", inputsOutputs);
-        for (Entry<String, Object> stringObjectEntry : inputsOutputs.entrySet()) {
-            final Object value = stringObjectEntry.getValue();
-            if (value instanceof Map || value instanceof List) {
-                final String key = stringObjectEntry.getKey();
-                if (key.equals(cwlID)) {
-                    if (value instanceof Map) {
-                        Map param = (Map<String, Object>)stringObjectEntry.getValue();
-                        handleOutputFile(fileMap, cwlID, param);
-                    } else {
-                        assert (value instanceof List);
-                        for (Object entry : (List)value) {
-                            if (entry instanceof Map) {
-                                handleOutputFile(fileMap, cwlID, (Map<String, Object>)entry);
-                            }
+        if (inputsOutputs.containsKey(cwlID)) {
+            Object jsonParameters = inputsOutputs.get(cwlID);
+            if (jsonParameters instanceof Map || jsonParameters instanceof List) {
+                if (jsonParameters instanceof Map) {
+                    Map param = (Map<String, Object>)jsonParameters;
+                    handleOutputFile(fileMap, cwlID, param);
+                } else {
+                    assert (jsonParameters instanceof List);
+                    for (Object entry : (List)jsonParameters) {
+                        if (entry instanceof Map) {
+                            handleOutputFile(fileMap, cwlID, (Map<String, Object>)entry);
                         }
                     }
                 }
+            } else {
+                System.out.println("WARNING: Output malformed for \"" + cwlID + "\" provisioning by default to working directory");
```

Comment:
One wrinkle, the ArgumentUtility is in the client module which depends on launcher and not vice versa. 
This is probably best done as a followup in the refactored 1.3 alpha https://github.com/ga4gh/dockstore/issues/796